### PR TITLE
test: update integration test function name and add e2e test

### PR DESCRIPTION
**Pull request by paulormnas**
<details>
  <summary>PR Checklist</summary>

### PR Structure

* [ ] This PR has reasonably narrow scope (if not, break it down into smaller PRs).
* [ ] This PR avoids mixing refactoring changes with feature changes (split into two PRs
  otherwise).
* [ ] This PR's title starts with name of package that is most changed in the PR, ex.
  `paymentservice.stellar`, or `all` or `doc` if the changes are broad or impact many
  packages.

### Thoroughness

* [x] This PR adds tests for the most critical parts of the new functionality or fixes.
</details>

### What

Create docker-compose files for e2e test

### Why

This an important update to maintain and improve software quality

### Known limitations

N/A

### CODE REVIEW

File: docker-compose.yaml [hunk 1]
```diff
@@ -11,6 +11,8 @@ services:
       - SECRET_DATA_PASSWORD=password
       - SECRET_PLATFORM_API_AUTH_SECRET=myAnchorToPlatformSecret
       - SECRET_CALLBACK_API_AUTH_SECRET=myPlatformToAnchorSecret
+      - SECRET_SEP24_INTERACTIVE_URL_JWT_SECRET=secret_sep24_interactive_url_jwt_secret
+      - SECRET_SEP24_MORE_INFO_URL_JWT_SECRET=secret_sep24_more_info_url_jwt_secret
```
Comment (paulormnas) [inline docker-compose.yaml:15]: @lijamie98 I added this two env-var to run the project locally. 
You said it was already defined inside `docker-compose/anchor-get-started/sep-24/docker-compose.yaml`. But I was unsure if I should import all that configuration, so I added only these two. 

If it is not right I can remove them.

Comment (lijamie98) [reply docker-compose.yaml:15]: This file is to be deprecated and deleted. We should have all docker-compose files saved under `docker-compose` folder.

File: Makefile [hunk 2]
```diff
@@ -26,31 +26,39 @@ docker-push-e2e-test:
 build-docker-compose-tests:
 	docker-compose -f integration-tests/docker-compose-configs/docker-compose.base.yaml build --no-cache
 
-run-e2e-test-all:
-	make run-e2e-test-default-config
-	make run-e2e-test-allowlist
-	make run-e2e-test-unique-address
-	make run-e2e-test-sep24-withdrawal
+run-integration-test-all:
```
Comment (lijamie98) [inline Makefile:29]: Thanks for renaming the targets. After merging this, please make sure the e2e tests work in the CI pipeline.

File: end-to-end-tests/Dockerfile [hunk 3]
```diff
@@ -2,9 +2,8 @@ FROM python:3.10-alpine
 
 WORKDIR /app
 
-COPY end-to-end-tests/requirements.txt requirements.txt
+COPY ./end-to-end-tests/. .
+RUN ls -la
```
Comment (lijamie98) [inline end-to-end-tests/Dockerfile:6]: Remove `RUN ls -la`?

File: end-to-end-tests/docker-compose.yaml [hunk 4]
```diff
@@ -0,0 +1,85 @@
+version: '2.4'
```
Comment (lijamie98) [inline end-to-end-tests/docker-compose.yaml:1]: Please move this file to the `$PROJECT_DIR/docker-compose` directory.

File: end-to-end-tests/docker-compose.yaml [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+    command: "--sep-server"
+    environment:
+      - STELLAR_ANCHOR_CONFIG=file:/config/anchor-platform-config.yaml
+      - SECRET_SEP10_SIGNING_SEED=${SECRET_SEP10_SIGNING_SEED?err}
+      - SECRET_SEP10_JWT_SECRET=secret
+      - SECRET_PLATFORM_API_AUTH_SECRET=myAnchorToPlatformSecret
+      - SECRET_CALLBACK_API_AUTH_SECRET=myPlatformToAnchorSecret
+      - SECRET_DATA_USERNAME=postgres
+      - SECRET_DATA_PASSWORD=password
+      - LOG_APPENDER=console_appender
```
Comment (lijamie98) [inline end-to-end-tests/docker-compose.yaml:17]: Is there a specific reason to set `LOG_APPENDER`?

File: end-to-end-tests/docker-compose.yaml [hunk 6]
```diff
@@ -0,0 +1,85 @@
+version: '2.4'
+services:
+  anchor-platform-server:
+    image: anchor-platform
+    build:
+      context: ../../
+      dockerfile: integration-tests/Dockerfile
```
Comment (lijamie98) [inline end-to-end-tests/docker-compose.yaml:7]: Does this mean we need to build docker image every time we run?

File: end-to-end-tests/docker-compose.yaml [hunk 7]
```diff
@@ -0,0 +1,85 @@
...
+      - ./anchor-platform-default-configs:/config_defaults
+    tmpfs:
+      - /config
+    depends_on:
+      - db
+      - kafka
+    extra_hosts:
+      - "host.docker.internal:host-gateway"
+
+  stellar-observer:
```
Comment (lijamie98) [inline end-to-end-tests/docker-compose.yaml:29]: The `stellar-observer` is also needed in [sep24 docker-compose file](https://github.com/stellar/java-stellar-anchor-sdk/blob/develop/docker-compose/anchor-get-started/sep24/docker-compose.yaml). 

Is there any way to share this configuration so that we don't have to duplicate the same configuration.

File: platform/src/test/kotlin/org/stellar/anchor/platform/config/Sep24ConfigTest.kt [hunk 8]
```diff
@@ -23,8 +23,8 @@ class Sep24ConfigTest {
   @BeforeEach
   fun setUp() {
     secretConfig = mockk()
-    every { secretConfig.sep24MoreInfoUrlJwtSecret } returns "interactive url jwt secret"
-    every { secretConfig.sep24InteractiveUrlJwtSecret } returns "more_info url jwt secret"
+    every { secretConfig.sep24MoreInfoUrlJwtSecret } returns "more_info url jwt secret"
```
Comment (lijamie98) [inline platform/src/test/kotlin/org/stellar/anchor/platform/config/Sep24ConfigTest.kt:26]: Good catch! 👍 

This is a bug. Please submit a different PR for this bug fix.

File: settings.gradle.kts [hunk 9]
```diff
@@ -16,5 +16,8 @@ include("kotlin-reference-server")
 /** Integration tests */
 include("integration-tests")
 
+/** E2E tests */
+include("end-to-end-tests")
```
Comment (lijamie98) [inline settings.gradle.kts:20]: We don't have the `build.gradle` under `end-to-end`. This may be confusing. I think we may add the sub-project when we have some Kotlin source code.

**Comment by lijamie98** [comment]
Please also update the PR description for `What`, `Why` and `Known limitations`.

File: end-to-end-tests/Dockerfile (hunk 3 above)
Comment (paulormnas) [reply end-to-end-tests/Dockerfile:6]: Done

File: end-to-end-tests/docker-compose.yaml (hunk 4 above)
Comment (paulormnas) [reply end-to-end-tests/docker-compose.yaml:1]: Done

File: end-to-end-tests/docker-compose.yaml (hunk 6 above)
Comment (paulormnas) [reply end-to-end-tests/docker-compose.yaml:7]: I removed that, but docker won't build a new image if `anchor-platform` was found. The `build` configs are used only in the case when the image is not found

File: end-to-end-tests/docker-compose.yaml (hunk 7 above)
Comment (paulormnas) [reply end-to-end-tests/docker-compose.yaml:29]: Updated the common-services file to include the stellar-observer.

File: platform/src/test/kotlin/org/stellar/anchor/platform/config/Sep24ConfigTest.kt (hunk 8 above)
Comment (paulormnas) [reply platform/src/test/kotlin/org/stellar/anchor/platform/config/Sep24ConfigTest.kt:26]: PR created and merged

File: settings.gradle.kts (hunk 9 above)
Comment (paulormnas) [reply settings.gradle.kts:20]: It makes sense. I'll remove it for now

File: end-to-end-tests/docker-compose.yaml [hunk 10]
```diff
@@ -0,0 +1,85 @@
+version: '2.4'
+services:
+  anchor-platform-server:
+    image: anchor-platform
+    build:
+      context: ../../
+      dockerfile: integration-tests/Dockerfile
+    command: "--sep-server"
+    environment:
+      - STELLAR_ANCHOR_CONFIG=file:/config/anchor-platform-config.yaml
```
[the remark this answers is not in the document]
Comment (paulormnas) [reply end-to-end-tests/docker-compose.yaml:10]: Done

File: end-to-end-tests/docker-compose.yaml [hunk 11]
```diff
@@ -0,0 +1,85 @@
...
+    command: "--stellar-observer"
+    environment:
+      - STELLAR_ANCHOR_CONFIG=file:/config/anchor-platform-config.yaml
+      - SECRET_SEP10_SIGNING_SEED=${SECRET_SEP10_SIGNING_SEED?err}
+      - SECRET_SEP10_JWT_SECRET=secret
+      - SECRET_PLATFORM_API_AUTH_SECRET=myAnchorToPlatformSecret
+      - SECRET_CALLBACK_API_AUTH_SECRET=myPlatformToAnchorSecret
+      - SECRET_DATA_USERNAME=postgres
+      - SECRET_DATA_PASSWORD=password
+      - LOG_APPENDER=console_appender
```
[the remark this answers is not in the document]
Comment (paulormnas) [reply end-to-end-tests/docker-compose.yaml:41]: I removed and the tests still working

File: end-to-end-tests/docker-compose.yaml (hunk 6 above)
Comment (paulormnas) [reply end-to-end-tests/docker-compose.yaml:7]: Besides that when I try to pull anchor image from Docker registry it fails because I don't have access 😢

**Comment by paulormnas** [comment]
@lijamie98 I saw something strange because first I thought that if the test script runs without errors means that all tests ran successfully. But after seeing the log below I thought the tests are not that clear. Maybe I may be wrong but its another thing to talk after the holiday.

```
end-to-end-tests-anchor-platform-server-1   | cp: cannot stat '/config_defaults/*': No such file or directory
.
.
.
end-to-end-tests-end-to-end-tests-1 exited with code 0
```
